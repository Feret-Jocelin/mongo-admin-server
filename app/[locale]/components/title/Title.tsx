import GenericButton from "@components/ui/button/GenericButton";
import SvgIcon from "@components/ui/icon/SvgIcon";

import styles from './title.module.scss';
import Checkbox from "../ui/inputs/checkbox/Checkbox";
import { useEffect, useState } from "react";

interface TitleProps {
  title: string;
  actions?: string[]; // search, refresh, add
  onClick?: (action: string) => void;
  withJsonSwitch?: boolean;
  showJson?: boolean;
  setShowJson?: (value: boolean) => void;
}

const Title = ({ title, actions, onClick, showJson, setShowJson, withJsonSwitch }: TitleProps) => {

  const handleJsonToggle = (e: React.ChangeEvent<HTMLInputElement>) => {
    setShowJson && setShowJson(e.target.checked);
  };

  const [themeFront, setThemeFront] = useState<any>('');


  useEffect(() => {

    const handleThemeChange = () => {
      const htmlElement = document.querySelector('html');

      if (htmlElement) {
        const themeAttributeHTML = htmlElement.getAttribute('data-theme');
        
        if(themeAttributeHTML === 'dark'){
          setThemeFront('dark');
        } else if(themeAttributeHTML === 'light'){
          setThemeFront('light');
        } else {
          setThemeFront('auto');
        }
      }
    }

    const observer = new MutationObserver(handleThemeChange)
    const htmlElement = document.querySelector('html');



    if (htmlElement) {
      observer.observe(htmlElement, { attributes: true });

      handleThemeChange();

      return () => {
        observer.disconnect();
      };
    }
  }, [])

  return (
    <section className={styles['title-container']}>
      <h1 className={styles['title-container__title']}>{title}</h1>
      <div className={styles['title-container__buttons']}>
        {actions?.map((action, index) => (
          <GenericButton
            key={index}
            outline
            center
            size="small"
            onClick={() => onClick && onClick(action)}
          >
            <SvgIcon icon_name={action} />
          </GenericButton>
        ))}
        {withJsonSwitch &&
          <Checkbox
            label="Afficher JSON"
            value={showJson ? showJson : false}
            onChange={handleJsonToggle}
            color={themeFront}
          />
        }
      </div>
    </section>
  );
}

export default Title;