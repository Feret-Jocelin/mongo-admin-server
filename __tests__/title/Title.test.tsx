/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import Title from '@/app/[locale]/components/title/Title';

describe('Title', () => {
    it('renders', () => {
      const handleClose = jest.fn();
      const { container } = render(
        <Title 
          title='Mock Title'
          isViewSearch
          searchPlaceholder='Mock placeholder'
          actions={['refresh','add']}
          searchValue='Rearch value'
          onClick={handleClose}
        />
      )
      expect(container).toMatchSnapshot();
    })
  })