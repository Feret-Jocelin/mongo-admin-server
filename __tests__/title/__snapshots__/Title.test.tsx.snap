// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders 1`] = `
<div>
  <section
    class="title-container"
  >
    <div
      class="title-container__header"
    >
      <h1
        class="title-container__title"
      >
        Mock Title
      </h1>
    </div>
    <div
      class="title-container__buttons"
    >
      <div
        class="input-container search-input"
      >
        <input
          class="input "
          placeholder="Mock placeholder"
          type="text"
          value="Rearch value"
        />
      </div>
      <button
        class="button small outline center"
        type="button"
      >
        <div
          class="undefined "
        />
      </button>
      <button
        class="button small outline center"
        type="button"
      >
        <div
          class="undefined "
        />
      </button>
    </div>
  </section>
</div>
`;
